gobblin-data-management/src/test/java/org/apache/gobblin/data/management/copy/writer/FileAwareInputStreamDataWriterTest.java

```diff
@@ -481,4 +523,11 @@ public void cleanup() {
     }
   }
 
-}
+  protected static class TestLocalFileSystem extends LocalFileSystem {
```

Comment:
why not encapsulate the `ConcurrentHashMap` here?  just add a method to access it for verification (copying-on-write would be most typical).